src/global/ValueId.h

```diff
@@ -78,23 +78,26 @@ class ValueId {
   /// `maxIndex`.
   struct IndexTooLargeException : public std::exception {
    private:
-    ad_utility::source_location _location;
-
-    T _tooBigValue;
-
     std::string _errorMessage;
 
    public:
-    IndexTooLargeException(
-        T tooBigValue,
-        ad_utility::source_location s = ad_utility::source_location::current())
-        : _location(s), _tooBigValue(tooBigValue) {
+    IndexTooLargeException(T tooBigValue,
+                           ad_utility::source_location s =
+                               ad_utility::source_location::current()) {
       std::stringstream exceptionMessage;
       exceptionMessage
-          << _location.file_name() << ", line " << _location.line()
-          << ": The given value " << _tooBigValue
+          << s.file_name() << ", line " << s.line() << ": The given value "
+          << tooBigValue
           << " is bigger than what the maxIndex of ValueId allows.";
       _errorMessage = exceptionMessage.str();
+      /*
+      TODO LLnk all files, that want to use this, with `absl::strings`. It
+      should has way better performance. Don't forget to add
+      `#include "absl/strings/str_cat.h"` to this file too.
+      _errorMessage = absl::StrCat(s.file_name(), ", line",
+        s.line(), ": The given value ", tooBigValue,
+        " is bigger than what the maxIndex of ValueId allows.");
+      */
```

Comment:
We now have a function in cmake (see the top-level-cmake) that is called `qlever_link_libraries` or `qlever_target_link_libraries` which we consistently use, so you should be able to use `absl::StrCat` now.

Reply:
Done, and thank you. (See [d02b8ff](https://github.com/ad-freiburg/qlever/pull/860/commits/d02b8ffabf3baffea4b28d5b56c45034b288aebb).)